Mathlib/Tactic/TryThis.lean

```diff
@@ -32,21 +33,28 @@ def addHaveSuggestion (ref : Syntax) (t? : Option Expr) (e : Expr) :
       `(tactic| let this := $estx)
   addSuggestion ref tac
 
-/-- Add a suggestion for `rw [h] at loc`. -/
-def addRewriteSuggestion (ref : Syntax) (e : Expr) (symm : Bool)
+open Lean.Parser.Tactic
+open Lean.Syntax
+
+/-- Add a suggestion for `rw [h₁, ← h₂] at loc`. -/
+def addRewriteSuggestion (ref : Syntax) (rules : List (Expr × Bool))
   (type? : Option Expr := none) (loc? : Option Expr := none)
   (origSpan? : Option Syntax := none) :
     TermElabM Unit := do
-  let estx ← delabToRefinableSyntax e
+  let rules_stx := TSepArray.ofElems <| ← rules.toArray.mapM fun ⟨e, symm⟩ => do
+    let t ← delabToRefinableSyntax e
+    if symm then `(rwRule| ← $t:term) else `(rwRule| $t:term)
   let tac ← do
-    let loc ← loc?.mapM fun loc => do `(Lean.Parser.Tactic.location| at $(← delab loc):term)
-    if symm then `(tactic| rw [← $estx] $(loc)?) else `(tactic| rw [$estx:term] $(loc)?)
+    let loc ← loc?.mapM fun loc => do `(location| at $(← delab loc):term)
+    `(tactic| rw [$rules_stx,*] $(loc)?)
 
   let mut tacMsg :=
+    let rulesMsg := MessageData.sbracket <| MessageData.joinSep
+      (rules.map fun ⟨e, symm⟩ => (if symm then "← " else "") ++ m!"{e}") ", "
     if let some loc := loc? then
-      if symm then m!"rw [← {e}] at {loc}" else m!"rw [{e}] at {loc}"
+      m!"rw {rulesMsg} at {loc}"
     else
-      if symm then m!"rw [← {e}]" else m!"rw [{e}]"
+      m!"rw {rulesMsg}"
```

Comment:
Is there a reason why it's creating a string representation from scratch rather than doing `let mut tacMsg := m!"{tac}"`? This string version isn't using the result of `delabToRefinableSyntax` for example, and I can't tell if that's intentional or an oversight.

Reply:
I think this was just an oversight. (Present in the code before this PR, I've just made the same mistake for each element of the list!)